File: 05072024/FrontEnd/src/app/dto/Utente.ts
import { Corso } from "./Corso";
import { Ruolo } from "./Ruolo";

export class Utente {
    private _id: number;
    private _nome: string;
    private _cognome: string;
    private _email: string;
    private _password: string;
    private _corsi: Corso[];
    private _ruoli: Ruolo[];

    constructor(id: number, nome: string, cognome: string, email: string, password: string, corsi: Corso[], ruoli: Ruolo[]) {
        this._id = id;
        this._nome = nome;
        this._cognome = cognome;
        this._email = email;
        this._password = password;
        this._corsi = corsi;
        this._ruoli = ruoli;
    }

    public get id(): number {
        return this._id;
    }

    public set id(id: number) {
        this._id = id;
    }

    public get nome(): string {
        return this._nome;
    }

    public set nome(nome: string) {
        this._nome = nome;
    }

    public get cognome(): string {
        return this._cognome;
    }

    public set cognome(cognome: string) {
        this._cognome = cognome;
    }

    public get email(): string {
        return this._email;
    }

    public set email(email: string) {
        this._email = email;
    }

    public get password(): string {
        return this._password;
    }

    public set password(password: string) {
        this._password = password;
    }

    public get corsi(): Corso[] {
        return this._corsi;
    }

    public set corsi(corsi: Corso[]) {
        this._corsi = corsi;
    }

    public get ruoli(): Ruolo[] {
        return this._ruoli;
    }

    public set ruoli(ruoli: Ruolo[]) {
        this._ruoli = ruoli;
    }
}

File: 28062024/restFinalFedericoRifugiato/src/main/java/com/federicorifugiato/model/Utente.java
package com.federicorifugiato.model;

import java.util.ArrayList;
import java.util.List;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinTable;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.JoinColumn;

@Entity
public class Utente {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ID_U")
    private int id;

    @Column(name = "Nome")
    private String nome;
    
    @Column(name = "Cognome")
    private String cognome;
    
    @Column(name = "email")
    private String email;
    
    @Column(name = "password")
    private String password;

	@ManyToMany(cascade = CascadeType.REFRESH, fetch = FetchType.EAGER)
    @JoinTable(
        name = "utenti_corsi",
        joinColumns = @JoinColumn(name = "FK_UC", referencedColumnName = "ID_U"),
        inverseJoinColumns = @JoinColumn(name = "FK_CU", referencedColumnName = "ID_C")
    )
    private List<Corso> corsi = new ArrayList<>();
    
    @ManyToMany(cascade = CascadeType.REFRESH, fetch = FetchType.EAGER)
    @JoinTable(
        name = "utente_ruolo",
        joinColumns = @JoinColumn(name = "FK_U", referencedColumnName = "ID_U"),
        inverseJoinColumns = @JoinColumn(name = "FK_R", referencedColumnName = "ID_G")
    )
    private List<Ruolo> ruoli = new ArrayList<>();
    
    public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getNome() {
		return nome;
	}

	public void setNome(String nome) {
		this.nome = nome;
	}

	public String getCognome() {
		return cognome;
	}

	public void setCognome(String cognome) {
		this.cognome = cognome;
	}

	public String getEmail() {
		return email;
	}

	public void setEmail(String email) {
		this.email = email;
	}

	public String getPassword() {
		return password;
	}

	public void setPassword(String password) {
		this.password = password;
	}

	public List<Corso> getCorsi() {
		return corsi;
	}

	public void setCorsi(List<Corso> corsi) {
		this.corsi = corsi;
	}

	public List<Ruolo> getRuoli() {
		return ruoli;
	}

	public void setRuoli(List<Ruolo> ruoli) {
		this.ruoli = ruoli;
	}
    
}
